### PR TITLE
fix fetching credentials

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards Node Pools.

### CODE REVIEW

File: service/controller/clusterapi/v28/machine_deployment_resource_set.go [hunk 1]
```diff
@@ -83,9 +87,26 @@ func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig)
 	}
 
 	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		var cr v1alpha1.Cluster
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v28/machine_deployment_resource_set.go]: I think this should be done in a resource and we should maybe drop the `initCtxFunc` thingy completely. I am not sure that thing was really useful at all.

Comment (kopiczko) [reply service/controller/clusterapi/v28/machine_deployment_resource_set.go]: I don't like passing information "up" the context in general, but I think we should address it other way so nothing against that if technically possible.

File: service/controller/clusterapi/v28/machine_deployment_resource_set.go [hunk 2]
```diff
@@ -83,9 +87,26 @@ func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig)
...
+		{
+			md, err := key.ToMachineDeployment(obj)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			id := key.WorkerClusterID(md)
```
Comment (kopiczko) [inline service/controller/clusterapi/v28/machine_deployment_resource_set.go]: Is this controller watching MachineDeployments? We usually were naming controllers after object type they were watching so I was pretty convinced it was watching cluster objects so far. But this makes sense. Do we have some diagrams showing relationships between the controllers and the Cluster API objects?

File: service/controller/clusterapi/v28/machine_deployment_resource_set.go (hunk 1 above)
Comment (xh3b4sd) [reply service/controller/clusterapi/v28/machine_deployment_resource_set.go]: What do you mean with passing up?

**Comment by kopiczko** [comment]
I did it! I approved after merging 🎉

File: service/controller/clusterapi/v28/machine_deployment_resource_set.go (hunk 2 above)
Comment (xh3b4sd) [reply service/controller/clusterapi/v28/machine_deployment_resource_set.go]: We named the controller `clusterapi`. That is not an object. There is probably some cleanup necessary at some point but we are still in "make it work". The picture below may be the best we have in terms of diagrams. Naming is already a bit outdated but it should transport the idea. 

Screenshot 2019-06-12 at 15 01 39

Comment (kopiczko) [reply service/controller/clusterapi/v28/machine_deployment_resource_set.go]: I understand thing modified here is the "NodePoolController" and we don't have "ClusterController" yet. Correct?

Comment (xh3b4sd) [reply service/controller/clusterapi/v28/machine_deployment_resource_set.go]: We have both controllers already. 

* https://github.com/giantswarm/aws-operator/blob/master/service/controller/clusterapi/cluster.go
* https://github.com/giantswarm/aws-operator/blob/master/service/controller/clusterapi/machine_deployment.go

They are just not in their own directory. We have three different controllers now in `clusterapi`. Two of them watch `Cluster` types. One of them watches `MachineDeployment` types.

Comment (kopiczko) [reply service/controller/clusterapi/v28/machine_deployment_resource_set.go]: That doesn't look right to me. So far we had controllers in separate directories. What is the reason to keep them together? Is it a short term solution?

Comment (xh3b4sd) [reply service/controller/clusterapi/v28/machine_deployment_resource_set.go]: > There is probably some cleanup necessary at some point but we are still in "make it work".

The reason is the transition we are in. We should get this straight at some point.

**Pull request merged**